src/transformers/modeling_gguf_pytorch_utils.py

```diff
@@ -321,6 +358,8 @@ def load_gguf_checkpoint(gguf_checkpoint_path, return_tensors=False):
                 f"From file name, cannot determine the number of parameters for {architecture} architecture"
             )
         model_size = m.group().strip("-")  # only keeps `7b`
+        if model_size == "40b":
+            parsed_parameters["config"]["new_decoder_architecture"] = True
```

Comment:
Can you add tests for the 40b even if we won't run them on the CI (add @skip) .

Reply:
In fact, I'm thinking about avoid using `model_size` extracted from file name, because some user fine-tuned model may use custom filename without "40b" explicitly. (I also removed the model_size logic in #34385)

Considering "new_decoder_architecture" having 2 layernorm for attn in decoder layer, I will prefer to check the existence of `attn_norm_2` to determine "new_decoder_architecture".

Reply:
```suggestion
        new_decoder_architecture = any("attn_norm_2" in tensor.name for tensor in reader.tensors)
        parsed_parameters["config"]["new_decoder_architecture"] = new_decoder_architecture
```

Reply:
cc @MekkCyber